clippy_lints/src/let_underscore.rs

```diff
@@ -191,15 +192,19 @@ impl<'tcx> LateLintPass<'tcx> for LetUnderscore {
             if local.pat.default_binding_modes && local.ty.is_none() {
                 // When `default_binding_modes` is true, the `let` keyword is present.
 
-				// Ignore function calls that return impl traits...
-				if let Some(init) = local.init &&
-				matches!(init.kind, ExprKind::Call(_, _) | ExprKind::MethodCall(_, _, _, _)) {
-					let expr_ty = cx.typeck_results().expr_ty(init);
-					if expr_ty.is_impl_trait() {
-						return;
-					}
-				}
-
+				if let Some(init) = local.init {
+                    // Ignore function calls that return impl traits...
+                    if matches!(init.kind, ExprKind::Call(_, _) | ExprKind::MethodCall(_, _, _, _)) {
+                        let expr_ty = cx.typeck_results().expr_ty(init);
+                        if expr_ty.is_impl_trait() {
+                            return;
+                        }
+                    }
+                    // Ignore if it is from a procedural macro...
+                    if is_from_proc_macro(cx, init) {
+                        return;
+                    }
```

Comment:
Tiny nit: This check is only executed if `local.init` is `Some`. However, the check should be done, whenever a `let_underscore_untyped` lint would be triggered. Could you move this below the if block, directly above the `span_lint_and_help`?

Note, in practice, I believe this doesn't matter, as I believe that locals without an init expression require a type. But moving it out of the block would be a bit cleaner either way.

Reply:
Hello @xFrednet! Thanks for suggesting this change. If makes a lot of sense. I have done it in the https://github.com/rust-lang/rust-clippy/pull/10775/commits/2d9d81fc1e881f4ccf84c0509ffc23c8086cdcc6 commit.

And my git was configured to sign the commits using a key whoose identity was my work email and not my personal email. Now I've changed it and it will be Verified!